types/redis/index.d.ts

```diff
@@ -32,30 +32,182 @@ export interface RetryStrategyOptions {
+     * or an error will be emitted if no callback is specified.
+     * @default true
+     */
     enable_offline_queue?: boolean;
     retry_max_delay?: number;
     connect_timeout?: number;
     max_attempts?: number;
+    /**
+     * If set to `true`, all commands that were unfulfilled while the connection is lost
+     * will be retried after the connection has been reestablished.
+     * Use this with caution if you use state altering commands (e.g. incr).
+     * This is especially useful if you use blocking commands.
+     * @default false
+     */
     retry_unfulfilled_commands?: boolean;
     auth_pass?: string;
+    /**
+     * If set, client will run Redis auth command on connect.
+     * Alias `auth_pass`.\
+     * **Note**: Node Redis < 2.5 must use `auth_pass`.
+     * @default null
+     */
     password?: string;
+    /**
+     * If set, client will run Redis **select** command on connect.
+     * @default null
+     */
     db?: string | number;
+    /**
+     * You can force using IPv6 if you set the family to **IPv6**.
+     * @see Node.js [net](https://nodejs.org/api/net.html)
+     * or [dns](https://nodejs.org/api/dns.html)
+     * modules on how to use the family type.
+     * @default IPv4
+     */
     family?: string;
+    /**
+     * If set to `true`, a client won't resubscribe after disconnecting.
+     * @default false
+     */
+    disable_resubscribing?: boolean;
+    /**
+     * Passing an object with renamed commands to use instead of the original functions.
+     * For example, if you renamed the command **KEYS** to "DO-NOT-USE"
+     * then the `rename_commands` object would be: { KEYS : "DO-NOT-USE" }.
+     * @see the [Redis security topics](http://redis.io/topics/security) for more info.
+     * @default null
+     */
     rename_commands?: { [command: string]: string } | null;
+    /**
+     * An object containing options to pass to
+     * [tls.connect](http://nodejs.org/api/tls.html#tls_tls_connect_port_host_options_callback)
+     * to set up a TLS connection to Redis
+     * (if, for example, it is set up to be accessible via a tunnel).
+     * @default null
+     */
     tls?: any;
+    /**
+     * A string used to prefix all used keys (e.g. namespace:test).
+     * Please be aware that the **keys** command will not be prefixed.
+     * The **keys** command has a "pattern" as argument and no key
+     * and it would be impossible to determine the existing keys in Redis if this would be prefixed.
+     * @default null
+     */
     prefix?: string;
+    /**
+     * A function that receives an options object as parameter including the retry `attempt`,
+     * the `total_retry_time` indicating how much time passed since the last time connected,
+     * the **error** why the connection was lost and the number of `times_connected` in total.
+     * If you return a number from this function, the retry will happen exactly after that time in milliseconds.
```

Comment:
Not sure "exactly" is a guarantee that can actually be made?

Reply:
I copied this from [README](https://github.com/NodeRedis/node-redis/blob/master/README.md).
Honestly, I am new to redis and I don't know what this option is doing. The best I can do is to remove the documentation from this option (retry_strategy).

Reply:
I think removing the word `exactly` fixes the concern raised here. Good work!

Reply:
I'd just remove the word "exactly" — they're basically making promises the runtime can't keep (due to the nature of the event loop + running on a preemptive multitasking OS).

Reply:
I removed the word **exactly** and pushed the change.